### ISSUE TITLE
False positive in `UnnecessaryParentheses`

### ISSUE THREAD

**Issue by igorwojda** [detekt/detekt#8668]
## Expected Behavior
Rule passes

## Observed Behavior
Rule Fails

## Steps to Reproduce
```kotlin
fun onInit(query: String? = (savedStateHandle[SAVED_QUERY_KEY] as? String) ?: DEFAULT_QUERY_NAME) {
        getAlbumList(query)
    }
```

## Your Environment

* Version of detekt used: 1.23.8
* Version of Gradle used (if applicable): `io.gitlab.arturbosch.detekt:detekt-gradle-plugin`
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository): https://github.com/igorwojda/android-showcase

**Comment by BraisGabin** [comment]
What's wrong with this?

```kotlin
fun onInit(query: String? = savedStateHandle[SAVED_QUERY_KEY] as? String ?: DEFAULT_QUERY_NAME) {
    getAlbumList(query)
}
```

**Comment by 3flex** [comment]
Closing due to lack of response. Can reopen if needed.
